### DISCUSSION TITLE
Error uploading image: [firebase_storage/object-not-found] No object exists at the desired reference.

### DISCUSSION THREAD

**Discussion by mynameismhmd** [firebase/flutterfire#10984] [Q&A]
here is my code how can I fix this`import 'dart:io';
import 'package:flutter/material.dart';
import 'package:image_picker/image_picker.dart';
import 'package:image_cropper/image_cropper.dart';
import 'package:firebase_storage/firebase_storage.dart' as firebase_storage;

class EditImagePage extends StatefulWidget {
  @override
  _EditImagePageState createState() => _EditImagePageState();
}

class _EditImagePageState extends State<EditImagePage> {
  File? _imageFile;
  String? _uploadedImageUrl;

  @override
  void initState() {
    super.initState();
    _pickImage();
  }

  void _pickImage() async {
    final pickedImage = await ImagePicker().getImage(source: ImageSource.gallery);
    if (pickedImage != null) {
      setState(() {
        _imageFile = File(pickedImage.path);
      });
    }
  }

  

  Future<void> _saveImage() async {
  if (_imageFile != null) {
    final fileName = '${DateTime.now().millisecondsSinceEpoch}.jpg';
    final destination = 'images/$fileName';

    try {
      // Upload the image file to Firebase Storage
       firebase_storage.FirebaseStorage.instance
          .ref(destination)
          .putFile(_imageFile!);

      // Retrieve the download URL after a delay of 2 seconds
      // Adjust the delay time as needed
      await Future.delayed(const Duration(seconds: 2));

      final uploadedImageUrl = await firebase_storage.FirebaseStorage.instance
          .ref(destination)
          .getDownloadURL();

      setState(() {
        _uploadedImageUrl = uploadedImageUrl;
      });

      // Save the uploaded image URL to Firestore or perform other actions
      // Firestore.instance.collection('images').add({'url': uploadedImageUrl});
    } catch (error) {
      // Handle any errors that occur during the upload process
      print('Error uploading image: $error');
    }
  }
}

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Edit Image'),
        actions: [
          IconButton(
            onPressed: _saveImage,
            icon: Icon(Icons.save),
          ),
        ],
      ),
      body: Column(
        children: [
          Expanded(
            child: _imageFile != null
                ? Image.file(_imageFile!)
                : Container(),
          ),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: [
              IconButton(
                onPressed: _pickImage,
                icon: Icon(Icons.photo_library),
              ),
              
            ],
          ),
          if (_uploadedImageUrl != null)
            Text('Uploaded Image URL: $_uploadedImageUrl'),
        ],
      ),
    );
  }
}
`

**Comment by sxjal** [comment]
Its weird kinda workaround but this actually seems to work for me when I am breaking the requests into different segments and awaited all of them. 

for example;

`final firebaseimageupload = FirebaseStorage.instance
            .ref()
            .child('user_images')
            .child(
                '${userCredentials.user!.uid}.jpg'); //could also use _firebase.currentUser!.uid

        await firebaseimageupload.putFile(_selectedimage!);

        final imageurl = await firebaseimageupload.getDownloadURL();
`
